src/Submissions/Params/CreateDetailParams.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+namespace Smartling\Submissions\Params;
+
+use Smartling\Parameters\BaseParameters;
+
+/**
+ * Class CreateDetailParams
+ * @package Submissions\Params
+ */
+class CreateDetailParams extends BaseParameters
+{
+
+    const DATE_TIME_FORMAT = "Y-m-d H:i:s";
+
+    /**
+     * @param string $submissionUid
+     * @return $this
+     */
+    public function setSubmissionUid($submissionUid)
+    {
+        $this->set('submission_uid', (string)$submissionUid);
+        return $this;
+    }
+
+    /**
+     * @param array $translationAssetId
+     * @return $this
+     */
+    public function setTranslationAssetId(array $translationAssetId = [])
+    {
+        $this->set('translation_asset_id', json_encode($translationAssetId));
+        return $this;
+    }
+
+    /**
+     * @param $targetLocale
+     * @return $this
+     */
+    public function setTargetLocale($targetLocale)
+    {
+        $this->set('target_locale', (string)$targetLocale);
+        return $this;
+    }
+
+    /**
+     * @param $state
+     * @return $this
+     */
+    public function setState($state)
+    {
+        $state = (string)$state;
+        if (in_array($state, SubmissionDetailsStates::$allowedStates, true)) {
+            $this->set('state', $state);
+        }
+        return $this;
```

Comment:
I believe we should throw a `UnexpectedValueException` exception in case condition result is false.